www/templates/html/Peoplefinder/Developers.tpl.php

```diff
@@ -1,174 +1,119 @@
-                                        case "xml":
-                                            $code = "xml";
-                                            break;
-                                        default:
-                                            $code = "html";
-                                    }
-                                    ?>
-                                    <pre class="code">
-                                        <code class="<?php echo $code; ?>"><?php echo htmlentities($result); ?></code>
-                                    </pre>
-                                </li>
-                            </ul>
-                        </div>
-                         <?php
-                     }
-                     ?>
+            </ul>
+            <div class="wdn_tabs_content">
+                <?php foreach ($resource->getAvailableFormats() as $format): ?>
 
-                </div>
-            </li>
-        </ul>
-    </div>
-
-</div>
-<div class="col right">
-    <div id='resources' class="zenbox primary" style="width:200px">
-        <h3>Directory API</h3>
-        <p>The following is a list of resources for Directory.</p>
-        <ul>
-            <?php
-            foreach ($context->resources as $resource) {
-                echo "<li><a href='?view=developers&resource=$resource'>$resource</a></li>";
-            }
-            ?>
-        </ul>
+                    <?php
+                    $url = UNL_Peoplefinder::addURLParams($resource->getExampleURI(), array('format' => $format));
+                    if (substr($url, 0, 2) == '//') {
+                        $url = 'http:' . $url;
+                    }
+                    $method_name = 'get' . ucfirst($format) . 'Properties';
+                    ?>
+                    <div id="<?php echo $format; ?>">
+                        <pre><code>GET <?php echo $url; ?></code></pre>
+                        <?php if (count($resource->$method_name())): ?>
+                            <h2>Resource Properties</h2>
+                            <table class="zentable neutral">
+                            <thead><tr><th>Property</th><th>Description</th></tr></thead>
+                            <tbody>
+                                <?php foreach ($resource->$method_name() as $property => $description): ?>
+                                    <tr>
+                                      <td><?php echo $property ?></td>
+                                      <td><?php echo $description ?></td>
+                                    </tr>
+                                <?php endforeach; ?>
+                            </tbody>
+                        </table>
+                        <?php endif; ?>
+                        <h2>Response</h2>
+                        <?php
+                        //Get the output.
+                        if (!$result = file_get_contents($url)) {
+                            $result = "Error getting file contents.";
```

Comment:
Well, this is much more complicated than I originally thought. You are correct that if the application was originally architected to support simulating requests, there would be little technical technical debt.

Problems with this application and implementing simulated requests.

1. Routes are not defined in the php layer. They are in the htaccess (apache layer). So translating a url into the correct information can get dicy. This isn't a terrible problem right now because all endpoints are documented as `service.php?view=...`, however I don't think we want to advertise that endpoint much longer. https://github.com/unl/UNL_Directory/blob/master/www/sample.htaccess#L4
2. There are two different controllers, which makes this a little more complicated than the bulletin example
3. Much of the work of setting up the output controller and handling different formats is done in index.php and not the output controller itself. https://github.com/unl/UNL_Directory/blob/master/www/index.php#L40

Solutions:
1. Duplicate a lot of logic from index.php which will lead to a great amount of technical debt and issues down the road as the application is updated but the documentation is not.
2. Refactor the application to improve routing design so that simulating requests are more feasible
3. Stick with `file_get_contents` but improve the error message.
4. Move documentation out of the application itself and into a CMS.

Option 3 is the easiest solution. Option 2 is the best, but would also require a good deal of work, and I feel it is out of scope of the original intent of this PR. We should avoid option 1 at all costs. Option 4 doesn't sound like a terrible idea, the only downside is that it won't auto-update when the application changes.

What do you think @kabel ?